lib/connection.js

```diff
@@ -189,7 +197,7 @@ Connection.prototype._receive = function (data) {
 
     this.offset = 0;
 
-    this.queue.shift().resolve(data.slice(4, length + 4));
+    this.queue.shift().resolve(Buffer.from(data.slice(4, length + 4)));
```

Comment:
If .then does not execute before the data buffer changes, it will get the wrong data. Hence deep copy needed.